pallets/messages/src/lib.rs

```diff
@@ -221,18 +221,16 @@ pub mod pallet {
 		/// * [`Error::UnsupportedCidVersion`] - CID version is not supported (V0)
 		/// * [`Error::InvalidCid`] - Unable to parse provided CID
 		///
-		#[pallet::weight(T::WeightInfo::add_ipfs_message(cid.len() as u32))]
+		#[pallet::weight(T::WeightInfo::add_ipfs_message())]
 		pub fn add_ipfs_message(
 			origin: OriginFor<T>,
 			#[pallet::compact] schema_id: SchemaId,
 			cid: Vec<u8>,
 			#[pallet::compact] payload_length: u32,
 		) -> DispatchResult {
 			let provider_key = ensure_signed(origin)?;
-			// validate_cid returns the binary CID representation. We'll use that in an upcoming PR, not now.
-			// For now we just store the input CID in the payload.
-			Self::validate_cid(&cid)?;
-			let payload_tuple: OffchainPayloadType = (cid, payload_length);
+			let cid_binary = Self::validate_cid(&cid)?;
```

Comment:
Quick question: do we know in which cases `cid` will be different from `cid_binary`?

Reply:
cid will always be an encoded string (base32, base64, etc). (We will only accept string-encoded CIDs). Binary will always be the decoded version of the input string. So they will always be different.